kaggle/api/kaggle_api_extended.py

```diff
@@ -716,13 +752,16 @@ def competitions_list(self,
             raise ValueError('Invalid sort_by specified. Valid options are ' +
                              str(self.valid_competition_sort_by))
 
-        competitions_list_result = self.process_response(
-            self.competitions_list_with_http_info(group=group or '',
-                                                  category=category or '',
-                                                  sort_by=sort_by or '',
-                                                  page=page,
-                                                  search=search or ''))
-        return [Competition(c) for c in competitions_list_result]
+        with self.build_kaggle_client() as kaggle:
+            request = ApiListCompetitionsRequest()
+            request.group = group
+            request.page = page
+            request.category = category
+            request.search = search
+            request.sort_by = sort_by

```

Comment:
Just noticing that these have `or` clauses in the original code, e.g. - `category or ''`.  I'm not sure if this is still needed, feel free to ignore if not.